tests/regression_tests/source_dlopen/test.py

```diff
@@ -0,0 +1,70 @@
+import openmc
+import pytest
+import os
+import glob
+
+from tests.testing_harness import PyAPITestHarness
+
+# compile the external source
+def compile_source():
+    # first one should be CMakelists in share/ 
+    files = glob.glob('../../../*/CMakeLists.txt')    
+    assert len(files) != 0
+
+    # copy the cmakefile
+    status = os.system('rm -rf build')
+    assert os.WEXITSTATUS(status) == 0
+    status = os.system('mkdir build ; cd build ; cp ../'+files[0]+' .')
+    assert os.WEXITSTATUS(status) == 0
+    status = os.system('cd build ; cmake -DSOURCE_FILES=../source_sampling.cpp ; make')
+    assert os.WEXITSTATUS(status) == 0
+    status = os.system('cp build/libsource.so .')
+    assert os.WEXITSTATUS(status) == 0
+    status = os.system('rm -rf build')
+    assert os.WEXITSTATUS(status) == 0
```

Comment:
This is all very unix-y. The test should only be run if the platform is unix itself. This can be done with a pytest marker:
```Python
import sys

@pytest.mark.skipif(sys.platform not in ('linux', 'darwin')
def test_dlopen_source():
    ...
```
Also the use of `libsource.so` would only work on linux. If you check for dylib on macOS, it should work there too. Something like:
```Python
suffix = 'dylib' if sys.platform == 'darwin' else 'so'
```

Reply:
I'm wondering if we could lump all of these commands into a bash script and call `os.system` once to reduce the verbosity and number of `assert` checks here.

Reply:
this is done